uktrade/data-hub-frontend
# No investment projects (new dashboard)

## Description of change

This work is around the new dashboard, in particular, when a new user has no investment projects.

## Test instructions

You will need to configure a few things to view the new dashboard, see PR #3248 for more details.

To change the results coming from the API edit [task.js](https://github.com/uktrade/data-hub-frontend/blob/feature/no-investment-projects/src/client/components/MyInvestmentProjects/tasks.js#L23)

```js
   // from 
  .then(({ data }) => data)
  // to
  .then(({ data }) => {
    return {
      count: 0,
      results: [],
    }
  })
```

Having written the [functional tests](https://github.com/uktrade/data-hub-frontend/blob/feature/no-investment-projects/test/functional/cypress/specs/dashboard-new/no-investment-projects-spec.js) it is difficult to hook them up as the sandbox mock API endpoint is becoming increasingly [unwieldy](https://github.com/uktrade/data-hub-frontend/blob/feature/no-investment-projects/test/sandbox/routes/v3/search/investment-project.js). For example, when making an API call to the search endpoint the body of the post contains the following:
```js
  {
    "limit": 10,
    "offset": 0,
    "adviser": "7d19d407-9aec-4d06-b190-d3f404627f21",
    "sortby": "created_on:desc"
  }
``` 
The problem is that we already return a set of results based on this payload, trying to return an empty array (no investment projects) with an identical payload is not going to work as we are unable to differentiate between the two. There are no other variables to play with other than changing the adviser id which is not ideal - **some feedback would be appreciated here**.

## Screenshots
<img width="1343" alt="no-investment-projects" src="https://user-images.githubusercontent.com/964268/112800678-e00e5900-9067-11eb-9491-8d36117a92c6.png">

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
